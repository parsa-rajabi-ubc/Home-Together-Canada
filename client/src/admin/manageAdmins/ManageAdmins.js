/**
 *
 * @Author:     Parsa Rajabi
 * @Created:    2021.3.12
 *
 * @Description: Manage Admins
 *
 */

import React from "react";
import PropTypes from "prop-types";
import SubmitButton from "../../common/forms/SubmitButton";
import GenericInput from "../../common/forms/GenericInput";

const ADMIN_TEXT = {
    TITLE: "Manage Admins",
    INFO: "Use the field below to search for members and provide them with admin privileges",
    CURRENT_ADMINS: "Current Website Admins",
}

function ManageAdmins(props) {
    const {
        setSearchUsername,
        searchUsernameError,
        currentAdmins,
        onSubmit
    } = props;


    const admins = currentAdmins.map(
        (person) =>
            <li key={person}>
                {person}
            </li>
    );

    // function to handle enter button
    const onKeyPress = (event) => {
        // checking if they pressed enter (ASCII Code)
        if (event.which === 13) {
            onSubmit(event);
        }
    }

    return (
        <div>
            <h3 className={"account-summary-info-header"}> {ADMIN_TEXT.TITLE} </h3>
            <p className="account-summary-info-text"> {ADMIN_TEXT.INFO} </p>

            {/*Search and Make Admins*/}
            <div className="p-4 flex mx-auto w-1/2 items-center">
                <GenericInput
                    className={`${searchUsernameError && "border-red-500"} input m-auto mr-20 py-2`}
                    placeholder="Search Member Username"
                    onChange={e => setSearchUsername(e.target.value)}
                    onKeyDown={onKeyPress}
                    inputType={"text"}
                />
                <SubmitButton
                    inputValue={"Make Admin"}
                    className="btn btn-green ml-10 w-2/3 py-2"
                    onClick={onSubmit}
                />
            </div>

            {/*Current Admins*/}
            <label className={"label text-lg mx-10"}> {ADMIN_TEXT.CURRENT_ADMINS} </label>
            <div className={"mx-20 my-3"}>{admins}</div>
        </div>
    );
}

ManageAdmins.propTypes = {
    onSubmit: PropTypes.func.isRequired,
    setSearchUsername: PropTypes.func.isRequired,
    searchUsernameError: PropTypes.bool,
    currentAdmins: PropTypes.array
};

export default ManageAdmins;