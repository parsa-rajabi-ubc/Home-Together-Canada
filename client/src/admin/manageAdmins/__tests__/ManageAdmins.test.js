/**
 * @Author:     Parsa Rajabi
 * @Created:    2021.3.12
 *
 * @Description: Manage Admins Test
 *
 */

import React from 'react';
import renderer from 'react-test-renderer'
import ManageAdmins from "../ManageAdmins";

describe('ManageAdmins', () => {
    describe('Container test', () => {
        it('should match snapshot test regardless of the props', () => {
            //given
            const props = {
                onSubmit: jest.fn(),
                setSearchUsername: jest.fn(),
                searchUsernameError: true,
                currentAdmins: ["Jack", "John"]
            }
            //when
            const component = renderer.create(<ManageAdmins {...props}/>);
            const tree = component.toJSON();
            //then
            expect(tree).toMatchSnapshot();
        });
    });
});