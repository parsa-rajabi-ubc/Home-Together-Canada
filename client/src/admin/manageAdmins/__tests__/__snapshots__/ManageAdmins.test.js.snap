// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageAdmins Container test should match snapshot test regardless of the props 1`] = `
<div>
  <h3
    className="account-summary-info-header"
  >
     
    Manage Admins
     
  </h3>
  <p
    className="account-summary-info-text"
  >
     
    Use the field below to search for members and provide them with admin privileges
     
  </p>
  <div
    className="p-4 flex mx-auto w-1/2 items-center"
  >
    <label>
      <label />
      <input
        className="border-red-500 input m-auto mr-20 py-2"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Search Member Username"
        type="text"
      />
    </label>
    <label>
      <input
        className="btn btn-green ml-10 w-2/3 py-2"
        onClick={[MockFunction]}
        type="submit"
        value="Make Admin"
      />
    </label>
  </div>
  <label
    className="label text-lg mx-10"
  >
     
    Current Website Admins
     
  </label>
  <div
    className="mx-20 my-3"
  >
    <li>
      Jack
    </li>
    <li>
      John
    </li>
  </div>
</div>
`;
